### PR TITLE
Use FlutterTestEnvironment for tests.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
- Use FlutterTestEnvironment to setUp/tearDown tests only when necessary. Significantly shortens test time.
- Consolidate "serviceManagerTests - restoring device-enabled extensions" group into a single test to avoid unnecessary setUp/tearDown.
- Use FlutterRunConfiguration throughout flutter_test_driver.dart.

### CODE REVIEW

File: test/inspector_controller_test.dart [hunk 1]
```diff
@@ -6,28 +6,23 @@
...
 import 'package:devtools/inspector/inspector_tree.dart';
 import 'package:devtools/inspector/inspector_text_styles.dart' as styles;
 import 'package:devtools/inspector/inspector_service.dart';
-import 'package:devtools/service_manager.dart';
 import 'package:devtools/ui/fake_flutter/fake_flutter.dart';
 import 'package:devtools/ui/flutter_html_shim.dart' as shim;
 import 'package:devtools/ui/icons.dart';
 import 'package:devtools/ui/material_icons.dart';
-import 'package:devtools/vm_service_wrapper.dart';
+import 'package:meta/meta.dart';
```
Comment (jacob314) [inline test/inspector_controller_test.dart]: keep packages outside of our project listed before packages from our project.

File: test/inspector_controller_test.dart [hunk 2]
```diff
@@ -551,7 +528,7 @@ void main() async {
       // TODO(jacobr): would be nice to have some tests that trigger a hot
       // reload that actually changes app state in a meaningful way.
 
-      await tearDownEnvironment();
+      await env.tearDownEnvironment(force: true);
```
Comment (jacob314) [inline test/inspector_controller_test.dart]: I think the force can be removed from this case?

File: test/inspector_service_test.dart [hunk 3]
```diff
@@ -228,11 +199,16 @@ void main() {
 
         await group.dispose();
 
-        await tearDownEnvironment();
+        await env.tearDownEnvironment();
       });
 
       test('render tree', () async {
-        await setupEnvironment(false);
+        await env.setupEnvironment(
```
Comment (jacob314) [inline test/inspector_service_test.dart]: even though it is more boilerplate, specify the config with each test.
That way if someone comments out a test or runs a single test the other tests will still run as expected

File: test/inspector_controller_test.dart (hunk 2 above)
Comment (kenzieschmoll) [reply test/inspector_controller_test.dart]: Done. added a forced tear down in tearDownAll

File: test/inspector_controller_test.dart (hunk 1 above)
Comment (kenzieschmoll) [reply test/inspector_controller_test.dart]: done.

File: test/inspector_service_test.dart (hunk 3 above)
Comment (kenzieschmoll) [reply test/inspector_service_test.dart]: Added runConfig param for each setupEnvironment call that should not use the default runConfig.

**Pull request merged**